Feature/DX-173-IPFS

W3stside opened the pull request:
- [x] Use uploaded Tokens.JSON in IPFS / defaultJSON as data to populate Tokens to choose
- [x] Add way to grab `defaultJSON`
- [x] Figure out what address to default to

W3stside committed:
added types for IPFS, updated @types, created IPFS API and HOC component

W3stside committed:
added types for IPFS, updated @types, created IPFS API and HOC component

W3stside committed:
hooked in some of IPFSHOC fns

W3stside committed:
removed state log & ghost div + fixed state props in TokenUpload

W3stside committed:
adding babel7 packages

W3stside committed:
comments

W3stside committed:
created Redux actions for previously internal IPFSHOC state + reducers + fixed minor CSS issue

W3stside committed:
Created containers for TokenPair, TokenPicker and TokenUpload

W3stside committed:
Actions, Reducers, Types, Styles, API, Story edited for Redux implementation of IPFSHOC:

1. Actions created and exported to index
2. Reducer created to save ipfs slice to state
3. Types edited to work
4. API created for IPFS integration
5. SCSS Styles and Story fix

W3stside committed:
Added fileContent and new actions to TokenUpload container

W3stside committed:
Transitioned IPFSHOC from internal State to Redux State

W3stside committed:
Added conditional rendering based off Redux state

W3stside commented:
Forgot to mention, WIP

W3stside committed:
Merge branch 'development_v2.0' of github.com:gnosis/dutch-exchange into feature/ipfs

Velenir commented on src/api/utils/index.ts:
```diff
@@ -13,6 +13,12 @@ export const windowLoaded = new Promise((accept, reject) => {
   }, false)
 })
 
+export const readFileUpload = (file: any) =>
```
Using native types this can be `export const readFileUpload = (file: File): Promise<ArrayBuffer> =>`

W3stside replied:
done

Velenir commented on src/types/index.d.ts:
```diff
@@ -110,14 +110,33 @@ export interface TokenOverlay {
   mod: TokenMod
 }
 
+export type FileBuffer = number[]
```
Why is it `number[]`. It's what you get after `FileReader.readAsArrayBuffer`? Then it's `ArrayBuffer` type

W3stside replied:
done

Velenir commented on src/types/index.d.ts:
```diff
@@ -110,14 +110,33 @@ export interface TokenOverlay {
 }
 
+export type FileBuffer = number[]
+export interface oFile {
```
This can be the native `File` type, no? Interface will remain the same

W3stside replied:
done

W3stside committed:
changes to Types as Dmitry suggest

W3stside committed:
changed import of oFile to use native File type

W3stside committed:
Merge branch 'develop' of github.com:gnosis/dutch-exchange into feature/ipfs

W3stside committed:
Feature/dx 218 state refresh (#224)

* created new API fns for calcing ongoingAuction state + types

* deleted OngoingAUctionsHOC + wrapped MenuAuctions in container and removed index table slot

* tweaked initialise.ts script to fire redux thunk action created in blockchian to check ongoingAUctions and other state based on timestamp + provider changes, created auctions actions

* created auctions reducer

* Small change to reducers to clean up and removed setup.js as unused

* updated API and api/types to calc MGN fee + get claimableTokens for all auctions

* updated Blockchain actions to add updateState and refresh store with new data on timestamp change

* updated components to take new data structure of ongoingAUctions, changed types, fixed minor CSS issue + created MenuFeeBalance container

* fixed typings in getSellerOngoingAuctions fn + made comments

* Reworked getSellerOngoingAuctions into Dmitry's edits + edited types and added provider check to only run with one provider

* added batchActions and changed in actions/blockchain.ts + added to store

* added typings in to remove @ts-ignore + cleaned up logic

W3stside committed:
Merge branch 'develop' of github.com:gnosis/dutch-exchange into feature/ipfs

W3stside committed:
fixed stories after prop changes + package.json / lock shenanigans

W3stside committed:
Removed internal state of defaultTokenList into Redux + made testing easier:
1. refactored udpateMainState test defaultTokenObject into api/apiTesting module to import around app
2. created actions/reducer for storing defaultTokenList and customTokenList in actions/tokenList
3. exported all actions from actions/index
4. added TokenOMG and TokenRDN to contracts api & types for use in FE

W3stside committed:
1. added conditional grabbing of TokenObject from localForage if there - else grabs from API (wip)
2. types

W3stside committed:
Created defaultTokenSelector to pass defaultTokenList from state into TokenOverlay components - replaces code2List[code] from earlier tests

W3stside committed:
Batched ipfsHash, defaultList + customList into function for conditional rendering and passed to necessary containers/components

Velenir committed:
set up user custom list logic

Velenir committed:
grabbed a couple tokens to use in a custom list

Velenir committed:
added isAddress function to web3 provider

Velenir committed:
added a check for custom token list for token validity

W3stside committed:
1. Changes to conditional rendering of TokenUpload
2. creation of CombinedTokenList

W3stside commented on src/containers/TokenUpload/index.ts:
```diff
@@ -1,32 +1,57 @@
 
 import { State } from 'types'
 
+import { batchActions } from 'redux-batched-actions'
```
in here as I felt too lazy to move it elsewhere in case we didnt like it -___-

Velenir committed:
simplified and fixed some types

Velenir commented on src/reducers/tokenList.ts:
```diff
@@ -1,20 +1,39 @@
+      combinedTokenList: [
+        ...state.defaultTokenList,
+        ...state.customTokenList,
+        ...action.payload.customTokenList,
```
this will always add to the current `combinedTokenList`. Try uploading the same file twice and see the duplicates appear.
Removing `...state.customTokenList` will fix that.

W3stside committed:
fixed small errors and changes to lists in reducer

Velenir committed:
customList from customHash

Velenir committed:
removed close button

Velenir committed:
dispatch customListHash to store

Velenir committed:
dispatch customListHash to store

Velenir committed:
memoizing token lists combinations

W3stside committed:
cleaned up and moved logic to correct places

W3stside committed:
fixed HTML error in TextSquare + added event.preventDefault in links

W3stside committed:
Fixed conditional rendering for tokenUpload + tokenPair + changed menuAuctions to show symbol

Velenir committed:
fixed some tslint errors

Velenir committed:
relaxed tslint rules

W3stside committed:
edited tokenList reducer to properly memoize and update tokenLists + changed batchAction for tokenList to move user to tokenPair page

W3stside committed:
divide bignumbers to represent Eth value in ongoingAuctions API function

W3stside committed:
Merge branch 'develop' into feature/ipfs
